virttest/qemu_vm.py

```diff
@@ -950,9 +950,12 @@ def add_spice_rhel5(devices, spice_params, port_range=(3100, 3199)):
             spice_help = ""
             if devices.has_option("spice-help"):
                 spice_help = commands.getoutput("%s -device \\?" % qemu_binary)
-            s_port = str(utils_misc.find_free_port(*port_range))
-            self.spice_options['spice_port'] = s_port
-            cmd += " port=%s" % s_port
+            if not self.is_alive():
```

Comment:
The thing here is that this works now, but in the future we might want to change the condition to skip evaluation from `is_alive` to something else and then this would break. Can I ask you one more change and then I promise I won't complain (about such issues) and that is to store those ports in `spice_options` and then update them in `create()` after retrieving the `spice_options`? Something like:

```diff
diff --git a/virttest/qemu_vm.py b/virttest/qemu_vm.py
index 97cd697..2349bde 100644
--- a/virttest/qemu_vm.py
+++ b/virttest/qemu_vm.py
@@ -2717,6 +2717,8 @@ class VM(virt_vm.BaseVM):
             try:
                 self.devices, self.spice_options = self.make_create_command()
                 self.update_vga_global_default(params, migration_mode)
+                self.spice_port = self.spice_options.get("spice_port")
+                self.spice_tls_port = self.spice_options.get("spice_tls_port")
                 logging.debug(self.devices.str_short())
                 logging.debug(self.devices.str_bus_short())
                 qemu_command = self.devices.cmdline()
```

PS: Don't forget to store them, in this patch you don't keep it in some branches.